src/raw/mod.rs

```diff
@@ -356,7 +358,7 @@ impl<T> Bucket<T> {
 }
 
 /// A raw hash table with an unsafe API.
-pub struct RawTable<T> {
+pub struct RawTable<T, A: AllocRef + Clone> {
```

Comment:
The `Drop` implementation for `RawTable` needs to call `free_buckets` which needs access to the allocator in order to free the bucket memory.

The `Drop` trait is also not allowed to have additional trait bounds compared to the struct itself.

I agree it would be nice to not have the trait bounds on the main `RawTable` struct, but as I see it we pretty much only have two options here:
1. Keep the bounds as is, the drop implementation will continue to do the freeing of the buckets.
2. Don't free the buckets automatically when dropping, require this to be done manually before the `RawTable` is dropped.

What do you think?

Reply:
Personally I am leaning towards option 2. The `RawTable` struct already has the concept of being uninitialized, so doing things this way will simplify the bounds in a lot of places, with only a small change to usage.

Reply:
You make a good point. I think it's fine to leave the bounds as they currently are.

Reply:
I don't see, why `Drop` needs `A: Clone`, could you elaborate on this? `drop` only calls `free_buckets`. It seems, that the `Clone`-bound could be removed in several places.